obspy/taup/seismic_phase.py

```diff
@@ -832,6 +832,31 @@ def add_to_branch(self, tau_model, start_branch, end_branch, is_p_wave,
                 self.min_ray_param,
                 tau_model.get_tau_branch(end_branch,
                                          is_p_wave).min_turn_ray_param)
+            # careful if the ray param cannot turn due to high slowness.
+            # Do not use these layers if their top is in high slowness for the
+            # given ray parameter and the bottom is not a critical reflection,
+            # rp > max rp in next branch
+            b_num = end_branch
+            while (b_num >= start_branch):
+                if (tau_model.s_mod.depth_in_high_slowness(
+                     tau_model.get_tau_branch(b_num, is_p_wave).top_depth,
+                     self.min_ray_param,
+                     is_p_wave) and
+                    (b_num + 1 >= len(tau_model.tau_branches[0])
+                     or self.min_ray_param <=
+                     tau_model.get_tau_branch(b_num+1,
+                                              is_p_wave
+                                              ).max_ray_param
+                     )):
```

Comment:
even if these conditions are single use, I think it would be better to have them singled out for readability.. even just the names for the condition would help a lot when coming back to this later  (naming just as an example as educated guesses)

```
while ...:
    slow_layer = tau_model...
    last_branch = b_num + 1 >= len(...
    ray_param_out_of_bounds = self.min_ray_param <= ...
    if (slow_layer and last_branch) or ray_param_out_of_bounds:
        ...
```

Reply:
The code style here was inherited from the java version. But I totally agree it would read better with the logic broken down into smaller steps, so I've done that. Once the tests have run take another look.